### PR TITLE
add documentation for validator

### PR DESCRIPTION
**Pull request by jihyeonyi**
### Summary
Validator documentation is added.

### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md)
- [x] I have updated the [documentation](
  https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have [linked related issues](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2020 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: docs/user_manual.md [hunk 1]
```diff
@@ -878,6 +879,157 @@ datum stats -p test_project
...
+        #  <label:str>: {
+        #     <attribute:str>: { <attr_value>: <mask_template>, }
+        #  }
+        'mask_distribution_in_dataset_item'] = <dict>,
+        # '(<item.id>, <item.subset>)': <mask/polygon count: int>
+    },
+    'validation_reports': <list>, #[ <validation_error_format>, ]
+    # validation_error_format = {
+    #   'anomaly_type': <str>,  # see datumaro/components/errors.py
+    #   'description': <str>,   # see datumaro/components/errors.py
```
Comment (zhiltsov-max) [inline docs/user_manual.md:981]: Who is supposed to read the code?

File: docs/user_manual.md [hunk 2]
```diff
@@ -878,6 +879,157 @@ datum stats -p test_project
 
 </details>
 
+
+### Validate project annotations
+
+This command inspects annotations based on the task type
```
Comment (zhiltsov-max) [inline docs/user_manual.md:885]:
```suggestion
This command inspects annotations with respect to the task type
```

File: docs/user_manual.md [hunk 3]
```diff
@@ -878,6 +879,157 @@ datum stats -p test_project
...
+
+The task types supported are `classification`, `detection`, and `segmentation`.
+
+The validation result contains
+- annotation statistics based on the task type
+- validation reports, such as
+    - items not having annotations
+    - items having undefined annotations
+    - imbalanced distribution in class/attributes
+    - too small or large values (items_far_from_mean)
```
Comment (zhiltsov-max) [inline docs/user_manual.md:896]: `items_far_from_mean` - it's not clear, what it means here.

File: docs/user_manual.md [hunk 4]
```diff
@@ -878,6 +879,157 @@ datum stats -p test_project
...
+``` bash
+{
+    'statistics': {
+        ## common statistics
+        'label_distribution': {
+            'defined_labels': <dict>,   # <label:str>: <count:int>
+            'undefined_labels': <dict>
+            #  <label:str>: {
+            #    'count': <int>,
+            #    'items_with_undefined_label': [(<item.id>, <item.subset>), ]
```
Comment (zhiltsov-max) [inline docs/user_manual.md:920]: Please be consistent in field definitions. 
It is unclear, what the dot means and what type the fields have.

File: docs/user_manual.md [hunk 5]
```diff
@@ -878,6 +879,157 @@ datum stats -p test_project
...
+        ## statistics for classification task
+        'items_with_multiple_labels': <list>, # [(<item.id>, <item.subset>), ]
+
+        ## statistics for detection task
+        'items_with_invalid_value': <dict>,
+        #  '(<item.id>, <item.subset>)' : {
+        #    <ann.id> : [ <property:str>, ]
+        #  }
+        #  properties: 'x', 'y', 'width', 'height',
+        #              'area(wxh)', 'ratio(w/h)', 'short', 'long'
```
Comment (zhiltsov-max) [inline docs/user_manual.md:951]: Clarify the meaning of `short` and `long`

File: docs/user_manual.md [hunk 6]
```diff
@@ -878,6 +879,157 @@ datum stats -p test_project
...
+        #  }
+        'mask_distribution_in_dataset_item'] = <dict>,
+        # '(<item.id>, <item.subset>)': <mask/polygon count: int>
+    },
+    'validation_reports': <list>, #[ <validation_error_format>, ]
+    # validation_error_format = {
+    #   'anomaly_type': <str>,  # see datumaro/components/errors.py
+    #   'description': <str>,   # see datumaro/components/errors.py
+    #   'severity': <str>, # 'warning' or 'error'    #
+    #   'item_id': <str>,  # when the validation target is a DatasetItem
```
Comment (zhiltsov-max) [inline docs/user_manual.md:983]: Validation target is a project or a dataset. Find a better phrasing.

**Review by zhiltsov-max — CHANGES_REQUESTED** [review]

File: docs/user_manual.md (hunk 1 above)
Comment (jihyeonyi) [reply docs/user_manual.md:981]: Do you mean the anomaly_type should be documented, instead of referring to the code?

Comment (jihyeonyi) [reply docs/user_manual.md:981]: I thought that this information is needed to parse the validation result.

File: docs/user_manual.md (hunk 3 above)
Comment (jihyeonyi) [reply docs/user_manual.md:896]: `items_far_from_mean` is a keyword of `numerical_stat_template`.
It means that a value in distribution is too big or too small to the avaerage value(=mean) of the distribution.
I think it's better to remove `(items_far_from_mean)`.

File: docs/user_manual.md (hunk 4 above)
Comment (jihyeonyi) [reply docs/user_manual.md:920]: `item` means  `DatasetItem`. I'll correct this.

File: docs/user_manual.md (hunk 5 above)
Comment (jihyeonyi) [reply docs/user_manual.md:951]: OK. I will.

File: docs/user_manual.md (hunk 6 above)
Comment (jihyeonyi) [reply docs/user_manual.md:983]: The validator inspects each DatasetItem and the distribution of them.
Here, is it assumed that the user doesn't know what the 'DatasetItem' is?

**Review by zhiltsov-max — APPROVED** [review]

**Pull request merged**